scipy/stats/_unuran/unuran_wrapper.pyx.templ

```diff
@@ -1888,3 +1888,338 @@ cdef class NaiveRatioUniforms(Method):
+
+        domain = _validate_domain(domain, dist)
+        if domain is not None:
+            if not np.isfinite(domain).all():
+                raise ValueError("`domain` must be finite.")
+        else:
+            if hasattr(dist, 'pmf'):
+                raise ValueError("`domain` must be provided when the "
+                                 "probability vector is not available.")
+
+        if guide_factor > 3:
+            msg = "guide_factor sizes larger than 3 are not recommended."
+            warnings.warn(msg, RuntimeWarning)
+
+        if guide_factor == 0:
+            msg = ("If the relative size (guide_factor) is set to 0, "
+                   "sequential search is used. However, this is not "
+                   "recommended, except in exceptional cases, since the "
+                   "discrete sequential search method has almost no setup and "
+                   "is thus faster.")
+            warnings.warn(msg, RuntimeWarning)
+
+        if hasattr(dist, 'pmf'):
+            # we assume the PMF accepts and return floats. So, we need
+            # to vectorize it to call with an array of points in the domain.
+            pmf = np.vectorize(dist.pmf)
+            k = np.arange(domain[0], domain[1]+1)
+            pv = pmf(k)
+            try:
+                pv_view = _validate_pv(pv)
+            except ValueError as err:
+                msg = "PMF returned invalid values: " + err.args[0]
+                raise ValueError(msg) from None
+        else:
+            pv_view = _validate_pv(dist)
+
+        return pv_view, domain
+
+    @cython.boundscheck(False)
+    @cython.wraparound(False)
+    cdef inline void _ppf(self, const double *u, double *out, size_t N):
+        cdef:
+            size_t i
+        for i in range(N):
+            out[i] = unur_dgt_eval_invcdf(self.rng, u[i])
+
+    def ppf(self, u):
+        """
+        ppf(u)
+
+        Approximated PPF of the given distribution.
+
+        Parameters
+        ----------
+        u : array_like
+            Qunatiles.
+
+        Returns
+        -------
+        ppf : array_like
+            Percentiles corresponding to given quantiles `u`.
+        """
+        u = np.asarray(u, dtype='d')
+        oshape = u.shape
+        u = u.ravel()
+
+        # UNU.RAN fills in ends of the support when u < 0 or u > 1 while
+        # SciPy fills in nans. Prefer SciPy behaviour.
+        cond0 = 0 <= u
+        cond1 = u <= 1
+        cond2 = cond0 & cond1
+        goodu = argsreduce(cond2, u)[0]
+        out = np.empty_like(u)
+
+        cdef double[::1] u_view = np.ascontiguousarray(goodu)
+        cdef double[::1] goodout = np.empty_like(u_view)
+
+        if cond2.any():
+            self._ppf(&u_view[0], &goodout[0], len(goodu))
+        np.place(out, cond2, goodout)
+        np.place(out, ~cond2, np.nan)
+
+        # UNU.RAN sets boundary at u = 0 to 0
+        # SciPy fills it with -1. Prefer SciPy behaviour
+        np.place(out, u == 0, -1)
```

Comment:
SciPy fills in `left_boundary - 1` for `u = 0`. So, if we have `dist = binom(4, 0.1, loc=10)`, it will give `9.0` for `dist.ppf(0)`. So, we should save the `domain` in a `self.domain` attribute in the `__cinit__` method and use `self.domain[0]-1` to fill in the `u = 0` case.

Reply:
So that would be: ``np.place(out, u == 0, self.domain[0]-1)``?

Reply:
> So that would be: `np.place(out, u == 0, self.domain[0]-1)`, but also storing the domain in self.domain

Yes, exactly.